File: dcp-solutions/2025-02-13/solution.rb
```ruby
def min_squares(n)
  dp = Array.new(n + 1, n + 1)
  dp[0] = 0
  (2..n).each do |i|
    (i * i..n).each do |j|
      dp[j] = [dp[j], dp[j - i * i] + 1].min
    end
  end
  dp[n]
end
```

File: dcp-solutions/2025-02-13/solution.c
```c
#include <stdio.h>

int min_squares(int n) {
    int dp[n + 1] = {0};
    dp[0] = 0;
    
    for (int i = 1; i <= n; i++) {
        int j = 1;
        while (j * j <= i) {
            dp[i] = dp[i] < dp[i - j * j] + 1 ? dp[i] : dp[i - j * j] + 1;
            j += 1;
        }
    }
    
    return dp[n];
}
```